WooCommerce/Classes/Yosemite/DefaultStoresManager.swift

```diff
@@ -424,19 +424,46 @@ private extension DefaultStoresManager {
 
     /// Synchronizes the order statuses, if possible.
     ///
-    func retrieveOrderStatus(with siteID: Int64) {
+    @MainActor
+    func retrieveOrderStatus(with siteID: Int64) async -> [OrderStatus]? {
         guard siteID != 0 else {
             // Just return if the siteID == 0 so we are not making extra requests
-            return
+            return nil
         }
 
-        let action = OrderStatusAction.retrieveOrderStatuses(siteID: siteID) { result in
-            if case let .failure(error) = result {
-                DDLogError("⛔️ Could not successfully fetch order statuses for siteID \(siteID): \(error)")
-            }
+        return await withCheckedContinuation { continuation in
+            dispatch(OrderStatusAction.retrieveOrderStatuses(siteID: siteID) { result in
+                switch result {
+                    case let .success(statuses):
+                        continuation.resume(returning: statuses)
+                    case let .failure(error):
+                        DDLogError("⛔️ Could not successfully fetch order statuses for siteID \(siteID): \(error)")
+                        continuation.resume(returning: nil)
+                }
+            })
         }
+    }
 
-        dispatch(action)
+    /// Synchronizes the number of products for site snapshot tracking.
+    ///
+    @MainActor
+    func retrieveNumberOfProducts(siteID: Int64) async -> Int64? {
+        guard siteID != 0 else {
+            // Just return if the siteID == 0 so we are not making extra requests
+            return nil
```

Comment:
Would this give us problems with application passwords, as they don't have a site id?

Reply:
The site ID for application password sites is `-1` according the code here:

https://github.com/woocommerce/woocommerce-ios/blob/f15bc1324d80210aeafbe213960f8592b0b547c4/WooCommerce/Classes/System/WooConstants.swift#L69-L71

Reply:
Just tested logging in to a JN site with an application password, and the event was logged as:

```
🔵 Tracked application_store_snapshot, properties: [AnyHashable("woocommerce-paypal-payments"): "not_installed", AnyHashable("woocommerce-gateway-stripe"): "not_installed", AnyHashable("blog_id"): -1, AnyHashable("woocommerce-payments"): "not_installed", AnyHashable("products_count"): 0, AnyHashable("was_ecommerce_trial"): false, AnyHashable("is_wpcom_store"): false, AnyHashable("plan"): "", AnyHashable("woocommerce-square"): "not_installed", AnyHashable("orders_count"): 0]
```